src/modules-lua/noit/module/ssh2.lua

```diff
@@ -117,30 +117,37 @@ function elapsed(check, name, starttime, endtime)
 end 
 
 local KEXALG = {
+  -- TODO: Support RFC 4419 DH group exchange.
+  -- Main difference is rather than fixed groups, client proposes min/max/pref
+  -- group size in bits. Server then picks a group that best matches client
+  -- request.
   -- 'diffie-hellman-group-exchange-sha256',
   -- 'diffie-hellman-group-exchange-sha1', 
   'diffie-hellman-group14-sha1',
+  'diffie-hellman-group16-sha512',
+  'diffie-hellman-group18-sha512',
   'diffie-hellman-group1-sha1'
 }
 
-local KEYALG = { 'ssh-rsa', 'ssh-dss' }
+local KEYALG = { 'ssh-rsa', 'ecdsa-sha2-nistp256', 'ssh-ed25519', 'ssh-dss' }
 
 local ENCALG = {
-  'aes128-cbc', 'aes192-cbc', 'aes256-cbc', 'aes128-ctr', 'aes192-ctr',
-  'aes256-ctr', 'rijndael128-cbc', 'rijndael192-cbc', 'rijndael256-cbc',
-  '3des-cbc', '3des-ecb', '3des-cfb', '3des-ofb', '3des-ctr',
-  'blowfish-cbc', 'blowfish-ecb', 'blowfish-cfb', 'blowfish-ofb',
-  'blowfish-ctr', 'twofish128-ctr', 'twofish128-cbc', 'twofish192-ctr',
-  'twofish192-cbc', 'twofish256-ctr', 'twofish256-cbc', 'twofish-cbc',
-  'twofish-ecb', 'twofish-cfb', 'twofish-ofb', 'cast128-cbc', 'cast128-ecb',
-  'cast128-cfb', 'cast128-ofb', 'idea-cbc', 'idea-ecb', 'idea-cfb',
-  'idea-ofb', 'arcfour', 'arcfour128', 'arcfour256' }
```

Comment:
again why are things removed instead of just added to here?

Reply:
I removed all of the ones that use obscure modes (modes other than CTR or CBC).  I have never seen an SSH server in the wild that offers them, and ciphers like CAST and IDEA are flat-out obsolete in 2017.